static/js/components/dialogs/CollectionFormDialog.js

```diff
@@ -76,28 +78,67 @@ class CollectionFormDialog extends React.Component {
     dispatch(uiActions.setAdminLists(event.target.value));
   };
 
+  validate = () => {
+    const { collectionForm } = this.props;
+    const validation = validateCollectionForm(this.getPayload(collectionForm));
+    this.handleError(validation);
+  };
+
+  getPayload = (collectionForm: CollectionFormState) => {
+    return {
+      title: collectionForm.title,
+      description: collectionForm.description,
+      view_lists: this.calculateListPermissionValue(collectionForm.viewChoice, collectionForm.viewLists),
```

Comment:
I know this is outside this PR but can you move `calculateListPermissionValue` outside this class, so it can be more easily tested?

Reply:
Already tested [via lines 85-104](https://github.com/mitodl/odl-video-service/blob/0d261ffdd5c2efa63d2afef47967b8ee120675c5/static/js/components/dialogs/CollectionFormDialog_test.js#L85-L104), actual return values would be wrong if function wasn't working correctly.